xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAEnumeratorHD.cpp

```diff
@@ -546,238 +546,277 @@ ComPtr<ID3D11VideoProcessorOutputView> CEnumeratorHD::CreateVideoProcessorOutput
-                                           : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+    const char* loc = av_chroma_location_name(inputArgs.chroma_location);
+    const std::string_view location = loc ? loc : "unknown";
 
-  const ProcessorConversions convTopLeft =
-      ListConversions(m_input_dxgi_format, std::vector<DXGI_COLOR_SPACE_TYPE>{inputCS},
-                      RenderingOutputFormats, std::vector<DXGI_COLOR_SPACE_TYPE>{destCS});
+    CLog::LogF(LOGDEBUG,
+               "no supported conversions. Trying alternative chroma siting {} for HDR source.",
+               loc);
 
-  CLog::LogF(LOGDEBUG, "BT.2020 input color spaces supported with SDR {} range output: {}: {}",
-             DX::Windowing()->UseLimitedColor() ? "limited" : "full",
-             DX::DXGIColorSpaceTypeToString(inputCS), !convTopLeft.empty() ? "yes" : "no");
+    conversions =
+        LogAndListConversions(m_input_dxgi_format, inputArgs, RenderingOutputFormats, outputArgs);
+  }
 
-  if (!convTopLeft.empty())
+  if (conversions.empty())
   {
-    if (CServiceBroker::GetLogging().IsLogLevelLogged(LOGDEBUG) &&
-        CServiceBroker::GetLogging().CanLogComponent(LOGVIDEO))
-    {
-      for (const auto& c : convTopLeft)
-        CLog::LogF(LOGDEBUG, "supported conversion: {}", c.ToString());
-    }
-
-    CLog::LogF(LOGDEBUG, "Preferred UHD Blu-Ray top left chroma siting will be used.");
-    return convTopLeft;
+    CLog::LogF(LOGWARNING, "no conversions supported.");
   }
-
-  // Try left chroma siting as fall back - half a pixel is not much difference at 4k
-  inputCS = DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_LEFT_P2020;
-  const ProcessorConversions convLeft =
-      ListConversions(m_input_dxgi_format, std::vector<DXGI_COLOR_SPACE_TYPE>{inputCS},
-                      RenderingOutputFormats, std::vector<DXGI_COLOR_SPACE_TYPE>{destCS});
-
-  CLog::LogF(LOGDEBUG, "BT.2020 input color spaces supported with SDR {} range output: {}: {}",
-             DX::Windowing()->UseLimitedColor() ? "limited" : "full",
-             DX::DXGIColorSpaceTypeToString(inputCS), !convLeft.empty() ? "yes" : "no");
-
-  if (!convLeft.empty())
+  else
   {
     if (CServiceBroker::GetLogging().IsLogLevelLogged(LOGDEBUG) &&
         CServiceBroker::GetLogging().CanLogComponent(LOGVIDEO))
     {
-      for (const auto& c : convLeft)
+      for (const auto& c : conversions)
         CLog::LogF(LOGDEBUG, "supported conversion: {}", c.ToString());
     }
-
-    CLog::LogF(LOGDEBUG, "Left chroma siting will be used.");
-    return convLeft;
   }
 
-  return {};
+  return conversions;
 }
 
-ProcessorConversions CEnumeratorHD::QuerySDRConversions(const bool isSourceFullRange,
-                                                        AVColorPrimaries colorPrimaries,
-                                                        AVColorTransferCharacteristic colorTransfer)
+ProcessorConversions CEnumeratorHD::LogAndListConversions(
+    DXGI_FORMAT inputFormat,
```

Comment:
Unused parameter `DXGI_FORMAT inputFormat` ?

Reply:
Absolutely. left over from copy/paste. I'll do one better, outputFormats was also not strictly needed, even if used.